### PR TITLE
NEW Allow database read-only replicas

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/11352

Requires API added in https://github.com/silverstripe/silverstripe-framework/pull/11379

### CODE REVIEW

File: composer.json
Comment (ChloeSartorelli) [inline composer.json]: No need to bump - the config will simply be unused in earlier versions.

Comment (emteknetnz) [reply composer.json]: Updated

**Review by GuySartorelli — APPROVED** [review]
LGTM will merge after framework PR is merged

**Comment by GuySartorelli** [comment]
CI failures are unrelated

**Pull request merged**
